core/amber/src/main/python/core/architecture/packaging/output_manager.py

```diff
@@ -56,13 +62,28 @@ def __init__(self, worker_id: str):
         }
         self._ports: typing.Dict[PortIdentity, WorkerPort] = dict()
         self._channels: typing.Dict[ChannelIdentity, Channel] = dict()
+        self._port_result_writers: typing.Dict[PortIdentity, PortResultWriter] = dict()
 
-    def add_output_port(self, port_id: PortIdentity, schema: Schema) -> None:
+    def add_output_port(
+        self, port_id: PortIdentity, schema: Schema, storage_uri: str
+    ) -> None:
         if port_id.id is None:
             port_id.id = 0
         if port_id.internal is None:
             port_id.internal = False
 
+        if storage_uri != "":
+            document: VirtualDocument[Tuple]
+            document, _ = DocumentFactory.open_document(storage_uri)
+            writer = document.writer(str(get_worker_index(self.worker_id)))
+            writer_thread = PortResultWriter(writer)
+            threading.Thread(
+                target=writer_thread.run,
+                daemon=True,
+                name=f"port_storage_writer_thread_{port_id}",
+            ).start()
```

Comment:
Just want to make sure that the parent thread of this writer thread is the CP (main loop) thread, instead of DP thread. because techinically, DP thread can be killed and restarted (in case of failure or reset, etc). And restarting DP thread should not kill the writer thread (because you used daemon = True, and I vaguely remember it means it will stick to the life cycle of the parent thread). You can assume the CP (main loop) thread never dies.

Reply:
All the methods are called in main loop outside of DP thread.

Reply:
The parent thread is the thread that calls the `thread.start()` of a child thread. As long as the parent thread is CP (mainloop) thread, we are fine.

Reply:
`thread.start()` is called in `OutputManager.add_output_port()`, which is called by `AssignPortHandler.assign_port()`